frontend/src/lib/components/SelectBox.tsx

```diff
@@ -0,0 +1,248 @@
+    }, [])
+    return (
+        <div ref={dropdownRef} className="action-filter-dropdown">
+            <Row style={{ height: '100%' }}>
+                <Col sm={14} style={{ borderRight: '1px solid rgba(0, 0, 0, 0.1)', maxHeight: '100%' }}>
+                    <Input
+                        placeholder="Search events"
+                        autoFocus
+                        onChange={(e) => {
+                            setSearch(e.target.value)
+                        }}
+                        style={{ width: '100%', borderRadius: 0 }}
+                    />
+                    <div className="search-list">
+                        {items.map((item) => (
+                            <>
+                                <SelectUnit
+                                    name={item.name}
+                                    dropdownLogic={dropdownLogic}
+                                    dataSource={item.dataSource}
+                                />
+                                <Divider />
+                            </>
+                        ))}
+                    </div>
+                </Col>
+                <Col sm={10} className="info-box">
+                    {RenderInfo && <RenderInfo item={selectedItem} />}
+                </Col>
+            </Row>
+        </div>
+    )
+}
+
+export function SelectUnit({
+    name,
+    dataSource,
+    dropdownLogic,
+}: {
+    name: string | JSX.Element
+    dataSource: SelectedItem[]
+    dropdownLogic: actionFilterDropdownLogicType
+}): JSX.Element {
+    const [isCollapsed, setIsCollapsed] = useState(false)
+    const { setSelectedItem, clickSelectedItem } = useActions(dropdownLogic)
+    const { selectedItem, search, blockMouseOver } = useValues(dropdownLogic)
+    const data = !search
+        ? dataSource
+        : new Fuse(dataSource, {
+              keys: ['name'],
+          })
+              .search(search)
+              .map((result) => result.item)
+    return (
+        <>
+            <span onClick={() => setIsCollapsed(!isCollapsed)}>
+                <h4 style={{ cursor: 'pointer', userSelect: 'none', padding: '4px 12px', marginBottom: 0 }}>
+                    {isCollapsed || data.length === 0 ? <RightOutlined /> : <DownOutlined />} {name}
+                    <span
+                        style={{ float: 'right', fontWeight: search && data.length > 0 ? 700 : 'normal' }}
+                        className="text-small"
+                    >
+                        {data.length} event{data.length !== 1 && 's'}
+                    </span>
+                </h4>
+            </span>
+            {!isCollapsed && data.length > 0 && (
+                <List
+                    size="small"
+                    bordered={false}
+                    dataSource={data || []}
+                    renderItem={(item: SelectedItem) => (
+                        <List.Item
+                            className={selectedItem.key === item.key && 'selected'}
```

Comment:
`&& 'selected'` is always truthy. Did you mean something else here?

Reply:
In react this will set 'selected' as the class name